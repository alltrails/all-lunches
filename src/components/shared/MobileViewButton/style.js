import styled from 'styled-components';
import TransparentButton from 'styles/components/TransparentButton';
import { type, colors } from 'styles';

export const Button = styled(TransparentButton)`
  position: fixed;
  display: flex;
  align-items: center;
  left: 50%;
  bottom: 3%;
  transform: translate(-50%, -50%);
  background: ${colors.green};
  color: ${colors.white};
  border-radius: 6px;
  padding: 10px 25px;
  box-shadow: 2px 3px 3px rgb(0 0 0 / 30%), 0 1px 2px rgb(0 0 0 / 30%);
  font-weight: ${type.weight.bold};
`;

export const Icon = styled.img`
  height: 16px;
  margin-left: 12px;
`;
