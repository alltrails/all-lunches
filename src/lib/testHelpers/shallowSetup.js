import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
