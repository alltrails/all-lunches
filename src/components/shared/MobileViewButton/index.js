import React from 'react';
import PropTypes from 'prop-types';

import MapIcon from 'assets/map-icon.svg';
import ListIcon from 'assets/list-icon.svg';

import { Icon, Button } from './style';

const MobileViewButton = ({ isMapViewEnabled, onClick }) => {
  const buttonText = isMapViewEnabled ? 'Map' : 'List';
  const icon = isMapViewEnabled ? MapIcon : ListIcon;

  return (
    <Button onClick={onClick}>
      {buttonText}
      <Icon src={icon} alt="" />
    </Button>
  );
};

MobileViewButton.propTypes = {
  isMapViewEnabled: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default MobileViewButton;
