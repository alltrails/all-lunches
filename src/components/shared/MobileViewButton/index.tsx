import React from 'react';

import MapIcon from 'assets/map-icon.svg';
import ListIcon from 'assets/list-icon.svg';

import { Icon, Button } from './style';

type MobileViewButtonProps = {
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
  isMapViewEnabled: boolean;
};

const MobileViewButton = ({ isMapViewEnabled, onClick }: MobileViewButtonProps) => {
  const buttonText = isMapViewEnabled ? 'List' : 'Map';
  const icon = isMapViewEnabled ? ListIcon : MapIcon;

  return (
    <Button onClick={onClick}>
      {buttonText}
      <Icon src={icon} alt="" />
    </Button>
  );
};

export default MobileViewButton;
