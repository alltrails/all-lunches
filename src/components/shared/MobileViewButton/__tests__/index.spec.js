/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import MobileViewButton from '../index';

describe('MobileViewButton', () => {
  const defaultProps = {
    isMapViewEnabled: false,
    onClick: () => {},
  };
  const setup = shallowSetup(MobileViewButton, defaultProps);

  it('renders the expected view', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when map view is enabled', () => {
    const { wrapper } = setup({ isMapViewEnabled: true });
    expect(wrapper).toMatchSnapshot();
  });
});
