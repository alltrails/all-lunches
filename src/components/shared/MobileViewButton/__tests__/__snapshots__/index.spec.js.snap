// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileViewButton renders the expected view 1`] = `
<style__Button
  onClick={[Function]}
>
  Map
  <style__Icon
    alt=""
    src="test-file-stub"
  />
</style__Button>
`;

exports[`MobileViewButton renders the expected view when map view is enabled 1`] = `
<style__Button
  onClick={[Function]}
>
  List
  <style__Icon
    alt=""
    src="test-file-stub"
  />
</style__Button>
`;
